src/App.js

```diff
@@ -1,18 +1,22 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import {
+  Route
+} from 'react-router-dom';
+
+import MainHeader from './components/MainHeader';
+
+import ViewHome from './views/Home';
+import ViewFavorites from './views/Favorites';
+import ViewDetails from './views/Details';
 
 class App extends Component {
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      <div>
+        <MainHeader />
+        <Route path="/gif/:gifId" component={ViewDetails} />
+        <Route exact path="/your-favorites" component={ViewFavorites} />
+        <Route exact path="/" component={ViewHome} />
```

Comment:
👍 correct use of exact, otherwise the `ViewHome` component will render along with the rest of the components. Another way to add exclusiveness to the routing is using the [Switch](https://reacttraining.com/react-router/web/api/Switch) component